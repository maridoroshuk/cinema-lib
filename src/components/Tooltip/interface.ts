export interface ITooltipProps {
  children: string;
}
