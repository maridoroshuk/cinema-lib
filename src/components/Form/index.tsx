import React from 'react';
import { Formik } from 'formik';

import { IFormProps } from './interface';
import { StyledForm } from './styles';

export const Form = <T extends {}>({
  children,
  initialValues,
  validate,
  onFormSubmit,
}: IFormProps<T>) => {
  return (
    <Formik
      initialValues={initialValues}
      validate={validate}
      onSubmit={onFormSubmit}
    >
      <StyledForm>{children}</StyledForm>
    </Formik>
  );
};
