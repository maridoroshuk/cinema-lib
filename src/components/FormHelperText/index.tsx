import React from 'react';

import { IFormHelperTextProps } from './interface';
import { Wrapper } from './styles';

export const FormHelperText = ({ message }: IFormHelperTextProps) => {
  return <Wrapper>{message}</Wrapper>;
};
