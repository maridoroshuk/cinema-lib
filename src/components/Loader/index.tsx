import React from 'react';
import ReactLoading from 'react-loading';
import { ILoaderProps } from './interface';
import { LoaderWrapper } from './styles';

export function Loader({ height, width, color }: ILoaderProps) {
  return (
    <LoaderWrapper>
      <ReactLoading type="spokes" color={color} height={height} width={width} />
    </LoaderWrapper>
  );
}
