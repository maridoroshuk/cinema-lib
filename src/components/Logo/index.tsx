import React from 'react';

import LogoImage from '@/assets/Logo.png';
import { getLogoSize } from '@/helpers/getLogoSize';

import { LogoStyled } from './styles';
import { ILogo } from './interface';

export const Logo = ({ size }: ILogo) => {
  const width = getLogoSize(size);
  return (
    <LogoStyled src={LogoImage} width={width} alt="company logo (Modsen)" />
  );
};
