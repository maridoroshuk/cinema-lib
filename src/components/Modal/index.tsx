import React from 'react';

import { IModal } from './interface';
import { Wrapper } from './styles';

export const Modal = ({ children }: IModal) => {
  return <Wrapper>{children}</Wrapper>;
};
