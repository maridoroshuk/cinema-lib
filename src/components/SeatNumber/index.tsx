import React from 'react';

import { ISeatProps } from '@/components/SeatNumber/interface';
import { StyledSeat } from '@/components/SeatNumber/styles';

export const SeatNumber = ({
  children,
  backgroundColor,
  index,
}: Partial<ISeatProps>) => {
  return (
    <StyledSeat backgroundColor={backgroundColor} seatIndex={index}>
      {children}
    </StyledSeat>
  );
};
