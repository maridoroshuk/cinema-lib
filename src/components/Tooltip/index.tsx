import React from 'react';

import { ITooltipProps } from './interface';
import { Text } from './styles';

export const Tooltip = ({ children }: ITooltipProps) => {
  return <Text>{children}</Text>;
};
