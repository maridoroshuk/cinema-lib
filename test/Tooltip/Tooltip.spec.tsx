import React from 'react';
import { render } from '@testing-library/react';

import { Tooltip } from '@/components/Tooltip';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('Tooltip', () => {
  it('renders', () => {
    const { container } = render(
      setupTestProvider(<Tooltip>Required</Tooltip>),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
