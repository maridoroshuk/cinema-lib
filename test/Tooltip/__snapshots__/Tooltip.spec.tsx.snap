// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders 1`] = `
.c0 {
  width: 100%;
  word-break: break-word;
  font-family: 'Inter',sans-serif;
  font-size: 20px;
  background-color: #313131;
  color: #FFFFFF;
  text-align: center;
  padding: 10px;
  border-radius: 5px;
  position: absolute;
  bottom: 100%;
  left: 0;
}

.c0::after {
  content: '';
  position: absolute;
  top: 100%;
  left: 50%;
  margin-left: -10px;
  border-width: 10px;
  border-style: solid;
  border-color: #313131 transparent transparent transparent;
}

<span
  class="c0"
>
  Required
</span>
`;
